File: paper_artifacts/experiments/llr8/artifacts/sources/oss120b-fortran-skills/fuse_diamond__submitted.f90
subroutine fuse_diamond_fp64(a, out, LEN_1D, workspace, workspace_size) bind(C)
  use iso_c_binding
  implicit none
  integer(c_int64_t), value, intent(in) :: LEN_1D
  real(c_double), intent(in) :: a(LEN_1D)
  real(c_double), intent(out) :: out(LEN_1D)
  type(c_ptr), value, intent(in) :: workspace
  integer(c_int64_t), value, intent(in) :: workspace_size
  integer(c_int64_t) :: i
  real(c_double) :: x
  !$omp parallel do simd schedule(static)
  do i = 1, LEN_1D
    x = a(i)
    out(i) = x * x * x * x - 1.0d0
  end do
end subroutine fuse_diamond_fp64

File: paper_artifacts/experiments/llr8/artifacts/sources/qwen38-c-skills/fuse_diamond__submitted.c
#include <stdint.h>
#include <omp.h>

void fuse_diamond_fp64(double *a, double *out, int64_t LEN_1D,
                       uint8_t *workspace, int64_t workspace_size)
{
    (void)workspace;
    (void)workspace_size;
    const double *__restrict av = a;
    double *__restrict o = out;

    #pragma omp parallel for simd schedule(static)
    for (int64_t i = 0; i < LEN_1D; ++i) {
        double t = av[i] * av[i];
        o[i] = (t + 1.0) * (t - 1.0);
    }
}
